src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -1101,10 +1383,23 @@ CHIP_ERROR Spake2p_P256_SHA256_HKDF_HMAC::FEGenerate(void * fe)
     CHIP_ERROR error = CHIP_NO_ERROR;
     int result       = 0;
 
+#if defined(MBEDTLS_USE_TINYCRYPT)
+    mbedtls_uecc_keypair keypair;
+
+    uECC_set_rng(&uecc_rng_wrapper);
```

Comment:
Why is this not done once in the platform init? Could you determine if a random generator was set? Otherwise, again, this is not reentrant to other threads making use of crypto APIs